geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/utils/GeoMessageSerializer.scala

```diff
@@ -95,6 +94,18 @@ object GeoMessageSerializer {
     new GeoMessageSerializer(sft, serializer, kryoSerializer, avroSerializer, version)
   }
 
+  def partition(numPartitions: Int, toKey: => Array[Byte]): Int = {
+    // use the feature id if available, otherwise (for clear) use random shard
```

Comment:
hmm, not sure. `clear` kind of breaks all the paradigms around kafka partitioning. since ordering isn't guaranteed between partitions, even a single consumer might get the clear out of order. we'd almost need to send a clear to _each_ partition, and update the logic to only clear features from that partition. but that seems kind of complicated... going forward, it will always go to the same partition (although it won't necessarily be the first), since `clear` is always an empty key and value and the default partitioner uses a hash algorithm. this method was copied from the old implementation, so i'm inclined to not change the behavior...